### ISSUE TITLE
False positive comparison-with-callable in Enum method

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#2757]
I see that there are a lot of open issues related to Enums, but I can't say whether the root cause is the same or not.

### Steps to reproduce

Run PyLint on the following code:
```py
from enum import Enum

class Colors(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3
    def __lt__(self, other):
        if self.__class__ is other.__class__:
            return self.value < other.value
        return NotImplemented

print(Colors.RED < Colors.BLUE)
print(Colors.RED.value < Colors.BLUE.value)
```

### Current behavior

The following warning is issued:
```
9:19: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)
```
This is reported for the `self.value < other.value` comparison.

### Expected behavior

No warning should be issued.

Note that when comparing the `value` attributes outside the class (last `print` from the test case), PyLint doesn't issue any warnings. So it seems like the inferred type of `value` is different between inside and outside the method.

### pylint --version output

```
pylint 2.2.2
astroid 2.1.0
Python 3.5.6 (default, Feb  1 2019, 11:07:37) 
```

**Comment by PCManticore** [comment]
Thanks @mthuurne for reporting the issue! Nope, it doesn't look related to the rest of the `enum` issues.

**Comment by RemiCardona** [comment]
looks a lot like the bug I opened a while back : #2306

**Comment by mthuurne** [comment]
It does look like it's a duplicate of that bug. We even used the same example enum :)

**Comment by PCManticore** [comment]
Thanks for chiming in @RemiCardona Let's close this one then.
